sway-core/src/parse_tree/declaration.rs

```diff
@@ -137,129 +129,87 @@ impl Declaration {
     pub(crate) fn parse_from_pair(
         decl: Pair<Rule>,
         config: Option<&BuildConfig>,
-    ) -> CompileResult<Self> {
+    ) -> CompileResult<Vec<Self>> {
```

Comment:
could you document in the docstring why this now returns a vector and how it relates to desugaring?